File: services/management-service/domain/model/group.go
package model

import (
	"time"
)

// Group 工作组/分组领域模型
type Group struct {
	ID          string            `json:"id"`
	OrgID       string            `json:"orgId"`
	Name        string            `json:"name"`
	Code        string            `json:"code"`
	Type        GroupType         `json:"type"`
	Description string            `json:"description"`
	ParentID    *string           `json:"parentId,omitempty"`
	LeaderID    *string           `json:"leaderId,omitempty"`
	Attributes  map[string]string `json:"attributes,omitempty"`
	Status      GroupStatus       `json:"status"`
	MemberCount int               `json:"memberCount"` // 成员数量
	CreatedAt   time.Time         `json:"createdAt"`
	UpdatedAt   time.Time         `json:"updatedAt"`
	DeletedAt   *time.Time        `json:"deletedAt,omitempty"`
}

// GroupType 分组类型
type GroupType string

const (
	GroupTypeDepartment GroupType = "department" // 部门
	GroupTypeTeam       GroupType = "team"       // 团队
	GroupTypeShift      GroupType = "shift"      // 班组
	GroupTypeProject    GroupType = "project"    // 项目组
	GroupTypeCustom     GroupType = "custom"     // 自定义
)

// GroupStatus 分组状态
type GroupStatus string

const (
	GroupStatusActive   GroupStatus = "active"   // 活跃
	GroupStatusInactive GroupStatus = "inactive" // 停用
	GroupStatusArchived GroupStatus = "archived" // 归档
)

// IsActive 是否活跃
func (g *Group) IsActive() bool {
	return g.Status == GroupStatusActive
}

// GroupMember 分组成员关系领域模型
type GroupMember struct {
	ID         uint64     `json:"id"`
	GroupID    string     `json:"groupId"`
	EmployeeID string     `json:"employeeId"`
	Role       string     `json:"role"`
	JoinedAt   time.Time  `json:"joinedAt"`
	LeftAt     *time.Time `json:"leftAt,omitempty"`
	CreatedAt  time.Time  `json:"createdAt"`
	UpdatedAt  time.Time  `json:"updatedAt"`
}

// IsActive 成员是否在组内
func (gm *GroupMember) IsActive() bool {
	return gm.LeftAt == nil
}

// GroupFilter 分组查询过滤器
type GroupFilter struct {
	OrgID    string
	Type     GroupType
	Status   GroupStatus
	ParentID *string
	Keyword  string // 按名称、编码模糊搜索
	Page     int
	PageSize int
}

// GroupListResult 分组列表结果
type GroupListResult struct {
	Items    []*Group `json:"items"`
	Total    int64    `json:"total"`
	Page     int      `json:"page"`
	PageSize int      `json:"page_size"`
}

// GroupWithMembers 带成员的分组
type GroupWithMembers struct {
	*Group
	Members     []*Employee `json:"members"`
	MemberCount int         `json:"memberCount"`
}

File: frontend/web/src/types/group.ts
// 分组相关类型定义

export type GroupType = 'department' | 'team' | 'project' | 'custom'

export interface Group {
  id: string
  org_node_id: string
  name: string
  code: string
  type: GroupType
  parent_id?: string
  member_count: number
  description?: string
  created_at: string
  updated_at: string
}

export interface GroupTreeNode {
  id: string
  name: string
  type: GroupType
  parent_id?: string
  member_count: number
  children: GroupTreeNode[]
}

export interface CreateGroupRequest {
  name: string
  code: string
  type: GroupType
  parent_id?: string
  description?: string
}

export interface UpdateGroupRequest {
  name?: string
  code?: string
  type?: GroupType
  parent_id?: string
  description?: string
}

export interface GroupMember {
  id: string
  employee_id: string
  employee_name: string
  employee_no?: string
  role?: string
  joined_at: string
}

// ==================== 常量 ====================

export const GROUP_TYPE_OPTIONS = [
  { label: '部门', value: 'department' as GroupType },
  { label: '团队', value: 'team' as GroupType },
  { label: '项目', value: 'project' as GroupType },
  { label: '自定义', value: 'custom' as GroupType },
] as const

export function getGroupTypeText(type: GroupType): string {
  const map: Record<GroupType, string> = {
    department: '部门',
    team: '团队',
    project: '项目',
    custom: '自定义',
  }
  return map[type] || type
}

export function getGroupTypeTagType(type: GroupType): string {
  const map: Record<GroupType, string> = {
    department: '',
    team: 'success',
    project: 'warning',
    custom: 'info',
  }
  return map[type] || 'info'
}

/** 构建分组树 */
export function buildGroupTree(groups: Group[]): GroupTreeNode[] {
  const map = new Map<string, GroupTreeNode>()
  const roots: GroupTreeNode[] = []

  groups.forEach((g) => {
    map.set(g.id, {
      id: g.id,
      name: g.name,
      type: g.type,
      parent_id: g.parent_id,
      member_count: g.member_count,
      children: [],
    })
  })

  groups.forEach((g) => {
    const node = map.get(g.id)!
    if (g.parent_id && map.has(g.parent_id)) {
      map.get(g.parent_id)!.children.push(node)
    }
    else {
      roots.push(node)
    }
  })

  return roots
}
